### PR TITLE
scb: Multi-agent tests

### PR DESCRIPTION
**Pull request by j-mueller**
* Add `Plutus.SCB.Effects.MultiAgent` - similar to `Wallet.Emulator.MultiAgent` in that it lets you run actions in the context of agents, but different in the effects that are allowed
* `Plutus.SCB.TestApp.runTestApp` looks a bit nicer now

### CODE REVIEW

File: plutus-scb/test/Plutus/SCB/Effects/ContractTest.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import           Language.Plutus.Contract.Servant              (initialResponse, runUpdate)
+import qualified Language.PlutusTx.Coordination.Contracts.Game as Contracts.Game
+
+-- | A mock/test handler for 'ContractEffect'
+handleContractTest ::
+    (Member (Error SCBError) effs)
+    => Eff (ContractEffect ': effs)
+    ~> Eff effs
+handleContractTest = interpret $ \case
+    InvokeContract (InitContract "game") ->
```
Comment (michaelpj) [inline plutus-scb/test/Plutus/SCB/Effects/ContractTest.hs:31]: Hm, are we just missing the ability to "install" contracts? Seems like we shouldn't have to hardcode them.

File: plutus-scb/test/Plutus/SCB/Effects/MultiAgent.hs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import           Wallet.Emulator.NodeClient      (NodeControlEffect)
+import qualified Wallet.Emulator.NodeClient      as NC
+import           Wallet.Emulator.SigningProcess  (SigningProcessControlEffect)
+import qualified Wallet.Emulator.SigningProcess  as SP
+import           Wallet.Emulator.Wallet          (Wallet, WalletState)
+import qualified Wallet.Emulator.Wallet          as Wallet
+
+-- $multiagent
+-- An SCB version of 'Wallet.Emulator.MultiAgent', with agent-specific states and actions on them.
+-- Agents are represented by the 'Wallet' type.
```
Comment (michaelpj) [inline plutus-scb/test/Plutus/SCB/Effects/MultiAgent.hs:70]: An agent corresponds to an SCB, right?

File: plutus-scb/test/Plutus/SCB/Effects/ContractTest.hs (hunk 1 above)
Comment (michaelpj) [reply plutus-scb/test/Plutus/SCB/Effects/ContractTest.hs:31]: Ah, it's the same as currently.

Comment (j-mueller) [reply plutus-scb/test/Plutus/SCB/Effects/ContractTest.hs:31]: Yeah it's the same as before, I'm working on improving the install situation atm

File: plutus-scb/test/Plutus/SCB/Effects/MultiAgent.hs (hunk 2 above)
Comment (j-mueller) [reply plutus-scb/test/Plutus/SCB/Effects/MultiAgent.hs:70]: Yeah good point. I'll add that to the comment.

**Pull request merged**